Minor QOL improvements I desired

b0dee opened the pull request:
Addressed a number of improvements I desired: 

- [x] Launch on start
- [x] Toggle scroll bar
- [x] Customise highlight group

Commit:
Enable always active scrollbar

Commit:
Additional triggers (window/buffer enter)

Commit:
Auto launch when entering buffer

Commit:
Auto launch when entering window

- Fixed issues with first hack
- Added ability to toggle and set default to existing behaviour so no
  changes noticed unless desired by user
- NOTE: Had to do BufWinEnter for first appearence when launching Vim
  and WinEnter for split windows as no argument split does not trigger
  BufWinEnter

Commit:
Add ability to change highlight of the scroll bar

Commit:
Add ability to toggle scroll bar adhoc

noscript commented:
Thanks for the PR. Could you please also add new variables and command to the documentation?

Commit:
Updated docs

Commit:
Updated docs

b0dee commented:
Documentation updated, thanks for accepting it and great work on the plugin!

Commit:
Fix scroll bar not showing at bottom of file until topline was EOF

b0dee commented:
Submitted another as noticed the scroll bar wouldn't reach end of window until topline == last line in file
It will now put it to bottom of screen the moment the last line of file is on the window (if it wasn't going to already, not sure if edge case but happened a bit in files I was testing on)

noscript commented on autoload/elevator.vim:
```diff
@@ -7,6 +7,16 @@ endif
   g:elevator#width = 1
 endif
+if !exists('g:elevator#show_on_enter')
+  g:elevator#width = v:false
```
This condition unnecessary. Besides it breaks the compilation as `g:elevator#width` is boolean.

b0dee replied:
Sorry - bug, should be g:elevator#show_on_enter = false
Fixed in next PR

noscript commented on autoload/elevator.vim:
```diff
@@ -7,6 +7,16 @@ endif
+endif
+
+if !exists('g:elevator#hidden')
+  g:elevator#hidden = v:false
```
Please use `false`.

b0dee replied:
Fixing in next PR

noscript commented on autoload/elevator.vim:
```diff
@@ -33,8 +43,18 @@ def S__calculate_scale(winid__a: number): float
 
+export def Toggle()
+  if g:elevator#hidden
+    g:elevator#hidden = v:false
```
`true`, `false`. Optionally `g:elevator#hidden = !g:elevator#hidden`

b0dee replied:
Neatened and used toggling boolean (!)
Fixes in next PR

noscript commented on doc/elevator.txt:
```diff
@@ -40,3 +41,38 @@ Example:
     g:elevator#width = 2
 <
+
+*g:elevator#highlight*                Set the highlight group for the scroll bar
```
Please put `.` at the end of sentences, for consistency with the rest of documentation.

b0dee replied:
Resolved in next PR

noscript commented on plugin/elevator.vim:
```diff
@@ -8,5 +8,12 @@ g:elevator_loaded = true
 augroup Elevator
-  autocmd WinScrolled * elevator#Show(expand('<amatch>')->str2nr())
+	autocmd!
+	if exists('g:elevator#show_on_enter') && g:elevator#show_on_enter == v:true
```
`if exists('g:elevator#show_on_enter') && g:elevator#show_on_enter`

b0dee replied:
Fixed in next PR

noscript commented on autoload/elevator.vim:
```diff
@@ -96,7 +119,12 @@ export def S__set_geometry()
+    popup_offset = wininfo.height - popup_height
+  else
+    popup_offset = S__clamp(((wininfo.topline - 1) * scale)->round()->float2nr(), 0, wininfo.height - popup_height)
+  endif
```
This change makes the scrollbar not move if buffer is scrolled below the last line, and then suddenly jump up when it's scrolled above the last line.

noscript commented:
> Submitted another as noticed the scroll bar wouldn't reach end of window until topline == last line in file It will now put it to bottom of screen the moment the last line of file is on the window (if it wasn't going to already, not sure if edge case but happened a bit in files I was testing on)

This is intended and follows the behavior of GVim scrollbars (`guioptions+=r`), because Vim allows to scroll the buffer past the last line.

Commit:
Fix comments (adding punctuation)

Commit:
Revert scrollbar EOF changes as this is expected Vim behaviour

Commit:
Fixing comments

- Change 'v:bool' to 'bool'
- Bug with g:elevator#show_on_enter variable not being set on init (used
  width variable causing conflict)
- General code cleanup

b0dee commented:
Ahh - learn something new everyday. All changes completed submitting PR now

Commit:
Debugging

Commit:
Revert "Debugging"

This reverts commit e01f4271736c105df8da7dccc79902a404da2f76.

Commit:
Hide scroll bar when line count is lower than window height

Commit:
Fixed returning wrong type

noscript commented:
I rebased and merged the remaining changes manually.
